File: MusicManagerAPI-client/src/main/java/music/manager/model/MusicTypeDTO.java
package music.manager.model;

import custom.validation.EnumValidator;
import music.manager.constants.MusicTypes;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;

/**
 * Created by ldalzotto on 25/04/2017.
 */
public class MusicTypeDTO {

    @NotNull
    @EnumValidator(enumClazz = MusicTypes.class)
    private String musicType;

    @NotNull
    @Pattern(regexp = "(^(https:\\/\\/)|(http:\\/\\/)).+")
    private String sourceUrl;

    public String getMusicType() {
        return musicType;
    }

    public void setMusicType(String musicType) {
        this.musicType = musicType;
    }

    public String getSourceUrl() {
        return sourceUrl;
    }

    public void setSourceUrl(String sourceUrl) {
        this.sourceUrl = sourceUrl;
    }
}

File: MusicManagerAPI-client/src/scala/com/ldz/music/manager/model/MusicTypeDTO.scala
package com.ldz.music.manager.model

import javax.validation.constraints.Pattern
import javax.validation.constraints.NotNull

import com.ldz.enumeration.ExternalMusicKey
import com.ldz.music.manager.constants.MusicTypes

import scala.annotation.meta.field
import scala.beans.BeanProperty

/**
  * Created by Loic on 20/06/2017.
*/
case class MusicTypeDTO(@(NotNull@field) @BeanProperty musicType: MusicTypes.Value,
                        @(NotNull@field) @(Pattern@field)(regexp = "(^(https:\\/\\/)|(http:\\/\\/)).+") @BeanProperty sourceUrl: String,
                       @BeanProperty musicParameters: Map[ExternalMusicKey.Value, String] ) {

  def this() {
    this(null, null, null)
  }

}
